src/dfx/src/commands/generate.rs

```diff
@@ -16,48 +17,66 @@ pub struct GenerateOpts {
 }
 
 pub fn exec(env: &dyn Environment, opts: GenerateOpts) -> DfxResult {
-    let env = create_agent_environment(env, None)?;
+    let env = create_agent_environment(env, None).context("Failed to create AgentEnvironment.")?;
```

Comment:
Another example where we aren't adding any useful context at this level; it would be better if `create_agent_environment` provided this.